File: src/main/java/io/curiousoft/ijudi/ordermanagement/validator/ValidOrderType.java
package io.curiousoft.ijudi.ordermanagement.validator;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target({ FIELD, TYPE})
@Retention(RUNTIME)
@Constraint(validatedBy = OrderTypeValidator.class)
@Documented
public @interface ValidOrderType {

    String message() default "Please supply shipping info for delivery or If you are paying in store, shipping should be null";

    Class<?>[] groups() default { };

    Class<? extends Payload>[] payload() default { };
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/validator/ValidOrderType.kt
package io.curiousoft.izinga.commons.validator

import java.lang.annotation.Documented
import javax.validation.*
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [OrderTypeValidator::class])
@Documented
annotation class ValidOrderType(
    val message: String = "Please supply shipping info for delivery or If you are paying in store, shipping should be null",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)